docs/AddNewOp.md

```diff
@@ -35,6 +35,9 @@ ambiguity, adding the links to refs if necessary
 6. type constraints about input and output tensors
 7. shape inference function Example:
 https://github.com/onnx/onnx/blob/master/onnx/defs/nn/defs.cc#L1183
+All operators MUST implement a shape inference function before it is accepted into ONNX.
```

Comment:
Do we want to enforce this? If yes, can we enforce this in the code rather than documentation (can happen in a separate PR)?

Reply:
For now - I vote that the key reviewers for ONNX ops (not a lot of people) have a common agenda - refrain from checking-in ops without shape inference and encourage contributors to implement them as part of code review process.

If we need to enforce such a thing, that obviously means we need to plug gaps (missing shape inference functions) along with the PR that enforces it...

Reply:
Asking shape inference for all the nodes, is it too much?

Also mention where to add the tests for shape inference? And provide a good example for shape inference?

Reply:
@pranavsharma this is a document on how to add a new op, so that I think it's ok to ask contributors to add shape inference function for new ops.

@houseroad @hariharans29 we may rephrase it a little bit to not a "MUST", but provide shape inference function when it's applicable and meaningful.

Reply:
Updated

Reply:
great. the updated one look good to me. @houseroad  sounds good to you?